RecoPixelVertexing/PixelTrackFitting/src/PixelTrackReconstruction.cc

```diff
@@ -77,7 +77,10 @@ void PixelTrackReconstruction::run(TracksWithTTRHs& tracks, edm::Event& ev, cons
           continue;
         }
       }
-
+      // all legacy tracks are "highPurity"
+      track->setQuality(reco::TrackBase::loose);
+      track->setQuality(reco::TrackBase::tight);
+      track->setQuality(reco::TrackBase::highPurity);
```

Comment:
I see why
```c++
track->setQuality(reco::TrackBase::highPurity);
```
but why also
```c++
track->setQuality(reco::TrackBase::loose);
track->setQuality(reco::TrackBase::tight);
```
?

Reply:
because is exclusive!
these are bits. (it's a legacy mess)

Reply:
ah... understood, sorry for the noise !
(maybe you could expand the comment just above so the other people who do not know the history are not tempted to clean it up, either...)

Reply:
Will add comment.
Still  I would be scared by anybody that modify this code and do not have enough deep knowledge of tracking trivia to be aware on how reco::Track quality is supposed to work. (and even worse that just go around and delete lines w/o asking why are there in first place).